### ISSUE TITLE
Trust line counted in /assets when not authorized

### ISSUE THREAD

**Issue by bartekn** [stellar/go#1933]
`horizon-cmp` found the following diff:

```diff
      {
        "_links": {
          "toml": {
            "href": "https://horizon.singularity.exchange/.well-known/stellar.toml"
          }
        },
        "asset_type": "credit_alphanum4",
        "asset_code": "BTC",
        "asset_issuer": "GB7ZHXCNHG5IMEHIS2MG54DR3QJIPAQIBFBKZDMCHSDLJYNJNZXKS4SO",
>        "paging_token": "BTC:GB7ZHXCNHG5IMEHIS2MG54DR3QJIPAQIBFBKZDMCHSDLJYNJNZXKS4SO",
<        "paging_token": "BTC_GB7ZHXCNHG5IMEHIS2MG54DR3QJIPAQIBFBKZDMCHSDLJYNJNZXKS4SO_credit_alphanum4",
        "amount": "90029991.0000000",
>        "num_accounts": 12,
<        "num_accounts": 9,
        "flags": {
          "auth_required": true,
          "auth_revocable": true,
          "auth_immutable": false
        }
      },
```
The first issue is that `paging_token` format has changed and I think it's fine because it's value shouldn't be interpreted in any way and just used.

The bigger problem is `num_accounts` value. It seems that the old code does not count not authorized (flag) trust lines to accounts with `auth_required=true`.

**Comment by ire-and-curses** [comment]
This is interesting. There was a [SE question](https://stellar.stackexchange.com/questions/2844/viewing-trustlines-for-authorization-required-assets-from-an-asset-issuer-perspe) yesterday where the author said:

> Ideally I would like to get a list of accounts that are authorized to hold your asset and also a list of accounts that have "applied" for a trustline to hold your asset via Horizon.

`num_accounts` seems a bit ambiguous in this context, since it could refer to either authorised trustlines, non-authorised trustlines, or both. It would be great if there was a way to disambiguate.

**Comment by abuiles** [comment]
@ire-and-curses  I think we could also add an extra param to `/accounts` - to return either authorized or unauthorized. It returns all by default. 

`?asset_authorized=0` & `?asset_authorized=1`

**Comment by bartekn** [comment]
Can you create a new issue for this? I think it's quite easy to do so maybe we could add it to 0.24.0 milestone. This issue is to just make sure `/assets` returns the same response as 0.22.2.

**Comment by bartekn** [comment]
Closed in #1936.
